lib/core/src/server/build-dev.js

```diff
@@ -337,25 +346,13 @@ export async function buildDevStandalone(options) {
             Fix the error below and restart storybook.
           `
         : stripIndents`
-            Broken build, fix the error below.
+            Broken build, fix the error above.
             You may need to refresh the browser.
           `
     );
     logger.line();
-    if (error instanceof Error) {
-      if (error.error) {
-        logger.error(error.error);
-      } else if (error.stats && error.stats.compilation.errors) {
-        error.stats.compilation.errors.forEach(e => logger.plain(e));
-      } else {
-        logger.error(error);
-      }
-
-      if (error.close) {
-        process.exit(1);
-      }
-    }
-    if (options.smokeTest) {
+    
+    if (options.smokeTest || error && error.close) {
```

Comment:
when is error.close set?

Reply:
Whenever dev-server has an unrecoverable error

It is set at this line: https://github.com/storybookjs/storybook/blob/next/lib/core/src/server/dev-server.js#L54

Reply:
do we want to shut down the server at this point? cc @ndelangen

Reply:
yeah probably

Reply:
@just-boris do you think you could do a follow up PR that closes the server?